homeassistant/components/geniushub/climate.py

```diff
@@ -41,7 +41,7 @@ async def async_setup_platform(
         [
             GeniusClimateZone(broker, z)
             for z in broker.client.zone_objs
-            if z.data["type"] in GH_ZONES
+            if 'type' in z.data and z.data["type"] in GH_ZONES
```

Comment:
This will fail CI. Does this work?
```suggestion
            if z.data.get("type") in GH_ZONES
```

Reply:
This should be a `z.data.get()`.

But the fix should really be in the client library.

@manzanotti ?